### PR TITLE
Make `process` as module

### LINKED ISSUE #161: `process` as module
there is the [Nodejs built-in `process` module](https://nodejs.org/api/process.html) available to use, and some packages (like [mongodb](https://github.com/mongodb/node-mongodb-native)) prefer using it instead of the global module.
The module implementation is pretty simple it [re-exports the global module](https://github.com/nodejs/node/blob/v21.6.2/lib/process.js)

would it be possible to do the same in `llrt`?

### PR DESCRIPTION
**Pull request by floydspace**
closes #161

Hey @richarddavison , please review carefully, I'm not as good in rust as in js and ts.
But I covered implementation with tests and it looks good to me,
if you see any improvements to make, pls let me know.

### CODE REVIEW

**Comment by floydspace** [comment]
I'm a bit concerned about repetition of code, that I have to explicitly remap all functions from global process to module process, but maybe it is the only way in Rust, I'm open to suggestions .

File: src/process.rs [hunk 1]
```diff
@@ -115,3 +122,49 @@ pub fn init(ctx: &Ctx<'_>) -> Result<()> {
...
+        declare.declare("default")?;
+        Ok(())
+    }
+
+    fn evaluate<'js>(ctx: &Ctx<'js>, exports: &mut Exports<'js>) -> Result<()> {
+        let globals = ctx.globals();
+        let process: Object = globals.get("process")?;
+
+        export_default(ctx, exports, |default| {
+            default.set("env", process.get::<&str, Value>("env"))?;
```
Comment (richarddavison) [inline src/process.rs:152]: We can iterate through the enumarable keys of the process object and then set them accordingly. That way we can get rid of the repeat:
 export_default(ctx, exports, |default| {
 
      for key in process.keys() {
         ...
      }
 }

File: tests/process.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import _process from "process";
+
+describe("process", () => {
+  it("should have a process env", () => {
+    assert.equal(_process.env.PWD, process.env.PWD);
```
Comment (richarddavison) [inline tests/process.test.ts:5]: Will this ENV exist for all platforms? Can we check for `_` or something else?

**Review by richarddavison — CHANGES_REQUESTED** [review]

**Comment by richarddavison** [comment]
Awesome first PR! You look like a true Rustacean  🦀

File: src/process.rs (hunk 1 above)
Comment (floydspace) [reply src/process.rs:152]: I wanted to do similar like it is done in `timers`.
However `timers` has all props as `Function` type
```
let function: Function = globals.get(func_name)?;
```
But for `process` it didn't work, bc all props are different types.
could you give me some snippet how to make it generic?

File: tests/process.test.ts (hunk 2 above)
Comment (floydspace) [reply tests/process.test.ts:5]: I can do assertion simply by `process.env` memory reference, should be sufficient.

File: src/process.rs (hunk 1 above)
Comment (floydspace) [reply src/process.rs]: I found better solution

File: src/process.rs [hunk 3]
```diff
@@ -115,3 +120,34 @@ pub fn init(ctx: &Ctx<'_>) -> Result<()> {
...
+
+        declare.declare("default")?;
+        Ok(())
+    }
+
+    fn evaluate<'js>(ctx: &Ctx<'js>, exports: &mut Exports<'js>) -> Result<()> {
+        let globals = ctx.globals();
+        let process: Object = globals.get("process")?;
+
+        exports.export("default", process)?;
```
Comment (richarddavison) [inline src/process.rs:149]: This won't allow you to access the named exports from process tho:
`import {env} from "process"`

**Review by richarddavison — CHANGES_REQUESTED** [review]

File: src/process.rs (hunk 3 above)
Comment (floydspace) [reply src/process.rs]: I addressed it, spot on.

**Review by richarddavison — APPROVED** [review]

**Pull request merged**